### PR TITLE
Fix TC test_detach_attach_2_data_volumes failures

### LINKED ISSUE #2211: Test "test_detach_attach_2_data_volumes" fails with AttributeError: 'OCS' object has no attribute 'attachments'
Test_disk_failures ->test_detach_attach_2_data_volumes fails   with AttributeError: 'OCS' object has no attribute 'attachments'

             worker_and_volume['volume'], nodes.detach_volume(worker_and_volume['worker'])
            )

tests/manage/z_cluster/nodes/test_disk_failures.py:127: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
ocs_ci/ocs/platform_nodes.py:485: in detach_volume
    self.aws.detach_volume(volume)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <ocs_ci.utility.aws.AWS object at 0x7fd5f97383c8>
volume = <ocs_ci.ocs.resources.ocs.OCS object at 0x7fd5fcff3160>, timeout = 120

    def detach_volume(self, volume, timeout=120):
        """
        Detach volume if attached
    
        Args:
            volume (Volume): The volume to delete
            timeout (int): Timeout in seconds for API calls
    
        Returns:
            Volume: ec2 Volume instance
    
        """
>       if volume.attachments:
E       AttributeError: 'OCS' object has no attribute 'attachments'

ocs_ci/utility/aws.py:351: AttributeError

Below run has more details:
https://ocs4-jenkins.rhev-ci-vms.eng.rdu2.redhat.com/job/qe-deploy-ocs-cluster/8137/testReport/junit/tests.manage.z_cluster.nodes.test_disk_failures/TestDiskFailures/test_detach_attach_2_data_volumes/

### PR DESCRIPTION
**Pull request by PrasadDesala**
- Fixes #2211 
-  Improvement in cleanup to handle node restart if the osd stays at CLBO state

Signed-off-by: Prasad Desala <tdesala@redhat.com>

### CODE REVIEW

**Comment by PrasadDesala** [comment]
https://ocs4-jenkins.rhev-ci-vms.eng.rdu2.redhat.com/job/qe-deploy-ocs-cluster/8318/console
Testcase Passed for some reason PR validation didn't update the result

**Comment by PrasadDesala** [comment]
@petr-balogh continuous-integration/travis-ci is at same state for a long time. 
Also in the PR validation job there were no failures but here the validation Job shows status as failed.
Are we seeing some issue? Can you please help.

**Comment by clacroix12** [comment]
> @petr-balogh continuous-integration/travis-ci is at same state for a long time.
> Also in the PR validation job there were no failures but here the validation Job shows status as failed.
> Are we seeing some issue? Can you please help.

The reason it's being reported as a failure is because the deploy job failed with this:
```
java.io.IOException: cannot find current thread
	at org.jenkinsci.plugins.workflow.cps.CpsStepContext.doGet(CpsStepContext.java:300)
	at org.jenkinsci.plugins.workflow.cps.CpsBodySubContext.doGet(CpsBodySubContext.java:88)
	at org.jenkinsci.plugins.workflow.support.DefaultStepContext.get(DefaultStepContext.java:67)
	at org.jenkinsci.plugins.credentialsbinding.impl.BindingStep$Callback.finished(BindingStep.java:254)
	at org.jenkinsci.plugins.credentialsbinding.impl.BindingStep$Execution2$Callback2.finished(BindingStep.java:162)
	at org.jenkinsci.plugins.workflow.steps.GeneralNonBlockingStepExecution$TailCall.lambda$onSuccess$0(GeneralNonBlockingStepExecution.java:140)
	at org.jenkinsci.plugins.workflow.steps.GeneralNonBlockingStepExecution.lambda$run$0(GeneralNonBlockingStepExecution.java:77)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
Finished: FAILURE
```
This error happened at the end up the pipeline, after the tests had ran and the cluster was destroyed. I'm going to mark this as verified since the deployment and tests actually passed.

Regarding the travis stuff, I've seen this a couple times where we can't get a response back and it just hangs forever. You can force a re-run by rebasing your branch on master and then force pushing to the remote. This will update the git hashes and trigger travis again. This will also invalidate any existing approvals, but it should hopefully get you some results from travis.

File: tests/manage/z_cluster/nodes/test_disk_failures.py [hunk 1]
```diff
@@ -45,6 +47,19 @@ def finalizer():
...
                 node.wait_for_nodes_status()
+
+            # Restart node if the osd stays at CLBO state
+            osd_pods_obj_list = get_osd_pods()
+            for pod in osd_pods_obj_list:
+                if pod.get().get(
+                    'status'
+                ).get(
+                    'containerStatuses'
+                )[0].get('state') == 'CrashLoopBackOff':
```
Comment (ebenahar) [inline tests/manage/z_cluster/nodes/test_disk_failures.py:58]: Please add `CrashLoopBackOff` to constants.py under pod statuses

**Comment by ebenahar** [comment]
@PrasadDesala can you please fix `test_detach_attach_worker_volume` to catch `in-use` EBS volume status as well?

**Comment by PrasadDesala** [comment]
> > @petr-balogh continuous-integration/travis-ci is at same state for a long time.
> > Also in the PR validation job there were no failures but here the validation Job shows status as failed.
> > Are we seeing some issue? Can you please help.
> 
> The reason it's being reported as a failure is because the deploy job failed with this:
> 
> ```
> java.io.IOException: cannot find current thread
> 	at org.jenkinsci.plugins.workflow.cps.CpsStepContext.doGet(CpsStepContext.java:300)
> 	at org.jenkinsci.plugins.workflow.cps.CpsBodySubContext.doGet(CpsBodySubContext.java:88)
> 	at org.jenkinsci.plugins.workflow.support.DefaultStepContext.get(DefaultStepContext.java:67)
> 	at org.jenkinsci.plugins.credentialsbinding.impl.BindingStep$Callback.finished(BindingStep.java:254)
> 	at org.jenkinsci.plugins.credentialsbinding.impl.BindingStep$Execution2$Callback2.finished(BindingStep.java:162)
> 	at org.jenkinsci.plugins.workflow.steps.GeneralNonBlockingStepExecution$TailCall.lambda$onSuccess$0(GeneralNonBlockingStepExecution.java:140)
> 	at org.jenkinsci.plugins.workflow.steps.GeneralNonBlockingStepExecution.lambda$run$0(GeneralNonBlockingStepExecution.java:77)
> 	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
> 	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
> 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
> 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
> 	at java.lang.Thread.run(Thread.java:748)
> Finished: FAILURE
> ```
> 
> This error happened at the end up the pipeline, after the tests had ran and the cluster was destroyed. I'm going to mark this as verified since the deployment and tests actually passed.
> 
> Regarding the travis stuff, I've seen this a couple times where we can't get a response back and it just hangs forever. You can force a re-run by rebasing your branch on master and then force pushing to the remote. This will update the git hashes and trigger travis again. This will also invalidate any existing approvals, but it should hopefully get you some results from travis.

Thank you Cody!

**Comment by PrasadDesala** [comment]
> @PrasadDesala can you please fix `test_detach_attach_worker_volume` to catch `in-use` EBS volume status as well?

Sure.

File: tests/manage/z_cluster/nodes/test_disk_failures.py (hunk 1 above)
Comment (PrasadDesala) [reply tests/manage/z_cluster/nodes/test_disk_failures.py]: Done

**Comment by PrasadDesala** [comment]
> > @PrasadDesala can you please fix `test_detach_attach_worker_volume` to catch `in-use` EBS volume status as well?
> 
> Sure.

Done

File: tests/manage/z_cluster/nodes/test_disk_failures.py [hunk 2]
```diff
@@ -123,16 +149,28 @@ def test_detach_attach_2_data_volumes(self, nodes, pvc_factory, pod_factory):
...
+            else:
+                """
+                Wait for worker volume to be re-attached automatically
+                to the node
+                """
+                assert nodes.wait_for_volume_attach(
+                    worker_and_volume['volume']
+                ), (f"Volume {worker_and_volume['volume']} failed to be "
+                    f"re-attached to a worker node"
+                    )
```
Comment (ebenahar) [inline tests/manage/z_cluster/nodes/test_disk_failures.py:173]: pretty much similar to lines 92-110 (https://github.com/red-hat-storage/ocs-ci/pull/2214/files#diff-d67bf88a5b02bb04cb2c3834aa727b19R92-R110). 
Please create a class method and call it from both places.

Comment (PrasadDesala) [reply tests/manage/z_cluster/nodes/test_disk_failures.py]: Done

**Comment by PrasadDesala** [comment]
The test case passed:
https://ocs4-jenkins.rhev-ci-vms.eng.rdu2.redhat.com/job/qe-deploy-ocs-cluster/8416/console

**Pull request merged**